### PR TITLE
don't rely on triggerStateListener to detect trigger errors

### PR DESCRIPTION
**Pull request by mck1117**
Check that various sync problems cause `getShaftSynchronized()` to return false.

### CODE REVIEW

**Comment by rusefillc** [comment]
once/if we are happy with this change can we merge test-only or mostly-test part first and then merge production code as separate merge?

**Comment by mck1117** [comment]
> once/if we are happy with this change can we merge test-only or mostly-test part first and then merge production code as separate merge?

yeah, I already had to break part off in to #4165

**Comment by rusefillc** [comment]
This PR has a functional change at the moment? Split one green PR into two green PRs?

File: unit_tests/tests/trigger/test_cam_vvt_input.cpp [hunk 1]
```diff
@@ -141,8 +140,8 @@ TEST(trigger, testNB2CamInput) {
 
 	int totalRevolutionCountBeforeVvtSync = 5;
 	// need to be out of VVT sync to see VVT sync in action
-	eth.fireRise(25);
-	eth.fireRise(25);
+	eth.fireRise(25 * 70 / 180);
```
Comment (mck1117) [inline unit_tests/tests/trigger/test_cam_vvt_input.cpp:143]: Hah! Mistake in the test caught by this change.

**Pull request merged**